app/lib/actions/katello/repository/import_upload.rb

```diff
@@ -2,17 +2,27 @@ module Actions
   module Katello
     module Repository
       class ImportUpload < Actions::EntryAction
-        def plan(repository, upload_id, options = {})
+        def plan(repository, upload_ids, options = {})
           action_subject(repository)
-          unit_key = options.fetch(:unit_Key, {})
+          unit_keys = options.fetch(:unit_keys, {})
           generate_metadata = options.fetch(:generate_metadata, true)
-          import_upload = plan_action(Pulp::Repository::ImportUpload,
-                                      pulp_id: repository.pulp_id,
-                                      unit_type_id: repository.unit_type_id,
-                                      unit_key: unit_key,
-                                      upload_id: upload_id)
+          sync_capsule = options.fetch(:sync_capsule, true)
+          sequence do
+            concurrence do
+              upload_ids.zip(unit_keys) do |upload_id, unit_key|
+                import_upload = plan_action(Pulp::Repository::ImportUpload,
+                                            pulp_id: repository.pulp_id,
+                                            unit_type_id: repository.unit_type_id,
+                                            unit_key: unit_key,
+                                            upload_id: upload_id)
 
-          plan_action(FinishUpload, repository, :dependency => import_upload.output, :generate_metadata => generate_metadata)
+                plan_action(FinishUpload, repository, :dependency => import_upload.output,
+                            generate_metadata: false)
+              end
+            end
+            plan_action(Katello::Repository::MetadataGenerate, repository) if generate_metadata
+            plan_action(Katello::Repository::CapsuleGenerateAndSync, repository) if sync_capsule
```

Comment:
we still want this to execute in the Run phase i think (via an async task)?  This is how everything else is done, so i'd assume we want to do it this way too.  Also right now if the capsule is down the upload fails entirely (planning errors out and we never get to the run phase), moving it there would fix that too.